### DISCUSSION TITLE
Next.js API routes throwing 404 error on Vercel deployment

### DISCUSSION THREAD

**Discussion by adomakins** [vercel/community#7439] [Help] [answered]
### Summary

Basically says it on the tin, but I have the file /pages/api/hello.js

It works perfectly on localhost, showing "Hello" as JSON in the browser...

But after deploying, the same path throws a 404 error.

Nothing in the logs either.

### Example

https://stellarr-45y3k9c5i-vitalize-digital.vercel.app/api/hello (latest deployment at the time of writing)

https://github.com/adomakins/stellarr (link to my public repo so you can see what it looks like)

### Steps to Reproduce

You can reproduce this problem by:

1. deploying

``` hello.js
// /pages/api/hello.js
export default async function handler(req, res) {
    console.log(`New event received`);
    res.status(200).json({ text: 'Hello' });
}

**Comment by pawlean** [comment]
Hi @adomakins,

I don't have access to that URL unfortunately. 

But can you check your Build Settings? See [relevant documentation](https://vercel.com/docs/deployments/configure-a-build#root-directory).

**Reply by adomakins** [reply, marked as answer]
Okay thank you! So I checked that out, changed the framework preset to Next.js, AND IT WORKED!

**Reply by pawlean** [reply]
Thanks for coming back to the community to share! Hopefully this will help community members too. 😄

**Reply by adomakins** [reply]
For anyone in the future, go Project → Settings → General → And make sure you've selected Next.js as the preset. 

My package.json had some random build command that I think was from npm init or something.

Get ridda allat, just let Vercel figure it out

**Reply by adomakins** [reply]
Here's some extra info I found on this issue

1. Make sure that you're using the right folder structure. If you're using the app folder then your api routes must be called route.js | route.ts in the right directory.

So essentially you'd have a file at /app/api/route.js which would be at the URL example.com/api

But if you use the pages directory in the root then you can have your endpoints at /api/example and it looks exactly like the folder structure.

2. If nothing here has solved your issues, here's another similar issue - https://github.com/orgs/vercel/discussions/4100#discussioncomment-10034935

[a comment by an automated account is not shown]
